File: flutter_baidu_yingyan_trace-1.0.0/lib/src/model/fence/delete_fence_option.dart
import 'package:flutter_baidu_yingyan_trace/src/model/base_option.dart';
import 'package:flutter_baidu_yingyan_trace/src/model/fence/fence_types.dart';

/// 删除围栏信息请求类
class DeleteFenceOption extends BaseOption {
  /// 围栏类型 （必传）
  FenceType ?fenceType;

  /// 要删除的地理围栏所监控的对象
  /// 可选。
  /// fenceIDs与monitoredObject共同决定了将被删除的地理围栏实体，规则如下：
  /// 若fenceIDs与monitoredObject同时设置，会删除指定的ID列表中，被监控对象为monitoredObject的地理围栏；
  /// 若设置了fenceIDs，但是没有设置monitoredObject，则会删除指定的ID列表对应的地理围栏；
  /// 若设置了monitoredObject，但是没有设置fenceIDs，则会删除所有被监控对象为monitoredObject的地理围栏；
  /// 若fenceIDs与monitoredObject都没有设置，则会删除所有的地理围栏。
  String ?monitoredPerson;

  /// 要删除的地理围栏的ID数组，若为null或空数组则删除监控对象上的所有地理围栏（必传）
  List<int> fenceIds;

  /// 构建删除本地围栏请求对象
  DeleteFenceOption.deleteLocalFenceOption(
      {required int tag,
      required int serviceId,
      required this.fenceIds,
      this.monitoredPerson})
      : assert(fenceIds != null),
        super(tag: tag, serviceId: serviceId) {
    fenceType = FenceType.local;
  }

  /// 构建删除服务端围栏请求对象
  DeleteFenceOption.deleteServerFenceOption(
      {required int tag,
      required int serviceId,
      required this.fenceIds,
      this.monitoredPerson})
      : assert(fenceIds != null),
        super(tag: tag, serviceId: serviceId) {
    fenceType = FenceType.server;
  }

  @override
  Map toMap() {
    Map map = super.toMap();
    map['fenceIds'] = this.fenceIds.map((e) => e).toList();
    map['monitoredPerson'] = this.monitoredPerson;
    map['fenceType'] = this.fenceType?.index;
    return map;
  }
}

File: flutter_baidu_yingyan_trace-1.0.0/android/src/main/java/com/baidu/flutter/trace/model/fence/DeleteFenceOption.java
package com.baidu.flutter.trace.model.fence;

import java.util.ArrayList;
import java.util.List;

import com.baidu.flutter.trace.model.BaseOption;
import com.baidu.trace.api.fence.DeleteFenceRequest;

/**
 * 删除围栏请求
 *
 * @author baidu
 */
public final class DeleteFenceOption extends BaseOption {

    /**
     * 监控对象
     */
    private String monitoredPerson;

    /**
     * 围栏编号列表
     */
    private List<Integer> fenceIds;

    /**
     * 围栏类型（本地围栏、服务端围栏）
     */
    private int fenceType;

    public String getMonitoredPerson() {
        return monitoredPerson;
    }

    /**
     * 设置监控对象
     *
     * @param monitoredPerson
     */
    public void setMonitoredPerson(String monitoredPerson) {
        this.monitoredPerson = monitoredPerson;
    }

    public List<Integer> getFenceIds() {
        return fenceIds;
    }

    /**
     * 设置围栏编号列表
     *
     * @param fenceIds
     */
    public void setFenceIds(List<Integer> fenceIds) {
        this.fenceIds = fenceIds;
    }

    public int getFenceType() {
        return fenceType;
    }
    
    public DeleteFenceOption() {
        super();
    }

    /**
     * 删除围栏请求
     *
     * @param tag             请求标识
     * @param serviceId       轨迹服务ID
     * @param monitoredPerson 监控对象
     * @param fenceIds        服务端围栏编号列表
     * @param fenceType       围栏类型
     */
    public DeleteFenceOption(int tag, long serviceId, String monitoredPerson,
                              List<Integer> fenceIds, int fenceType) {
        super(tag, serviceId);
        this.monitoredPerson = monitoredPerson;
        this.fenceIds = fenceIds;
        this.fenceType = fenceType;
    }

    public DeleteFenceRequest toDeleteFenceRequest() {
        DeleteFenceRequest deleteFenceRequest = null;
        if (fenceIds != null) {
            List<Long> fenceIdList = new ArrayList<>();
            for (Integer fenceId : fenceIds) {
                fenceIdList.add(fenceId.longValue());
            }
            if (fenceType == FenceType.local.ordinal()) {
                deleteFenceRequest = com.baidu.trace.api.fence.DeleteFenceRequest
                        .buildLocalRequest(tag, serviceId, monitoredPerson, fenceIdList);
            } else if (fenceType == FenceType.server.ordinal()) {
                deleteFenceRequest = com.baidu.trace.api.fence.DeleteFenceRequest
                        .buildServerRequest(tag, serviceId, monitoredPerson, fenceIdList);
            }
        }
        
        return deleteFenceRequest;
    }

    @Override
    public String toString() {
        return "DeleteFenceRequest [tag=" + tag + ", serviceId=" + serviceId + ", monitoredPerson="
                + monitoredPerson
                + ", fenceIds=" + fenceIds + ", fenceType=" + fenceType + "]";
    }

}
